main.py

```diff
@@ -203,175 +240,267 @@ async def spc_outlook(
     buffer = io.StringIO(file_content)
 
-    # Send the file
-    await processing_msg.delete()
-    file = File(fp=buffer, filename=f"{city}_{state}_{start_date}_{end_date}.txt")
-    await ctx.send(
+    # Send the final file and message using the existing followup
+    file = discord.File(
+        fp=buffer, filename=f"{city}_{state}_{start_date}_{end_date}.txt"
+    )
+    await interaction.followup.send(
         content=f"SPC Outlook for {city}, {state} (found {len(filtered_outlooks)} results)",
         file=file,
     )
 
 
-@client.command(
+currentDate = date.today().strftime("%B %d, %Y")
+
+
+@client.tree.command(
     name="fetch",
-    help="Fetches the latest Nadocast images. \n Usage: $fetch <params> \n Allowed params: sig, tor, wind, hail\n Examples: `$fetch tor`, `$fetch sig tor`",
+    description="Nadocast images for a given date, type, model, and zulu time. `/fetch March 1, 2024 tor 2024 12`",
 )
-async def fetch(ctx, *args) -> None:
-
-    await log("DEBUG: Fetch command called with args:", ",".join(args))
+@app_commands.describe(
+    date="Date (e.g. March 15, 2025)",
+    param="Type (tor, wind, hail, sig, life)",
+    models="Model (2024, 2024abs, 2022, 2022abs, blank for all) [optional]",
+    zulu="Zulu hour (0, 12, or 18) [optional]",
+)
+async def fetch(
+    interaction: discord.Interaction,
+    param: str,
+    date: str = currentDate,
+    models: str = "2022",
+    zulu: str = None,
+) -> None:
+    try:
+        await log(
+            "DEBUG: Fetch command called with params:",
+            date,
+            str(param),
+            str(models),
+            str(zulu),
+        )
+    except Exception as e:
+        await interaction.response.send_message(f"Log error: {e}", ephemeral=True)
+        return
 
     cooldown = cooldowns["fetch"]
-
-    # TODO: This try/except block is hardcoded with the params, needs fixing + adding the life risk param. Currently works and is not urgent.
     allowed_params = ["sig", "life", "tor", "wind", "hail"]
+    allowed_zulu = ["0", "12", "18"]
+    allowed_models = ["2024", "2024abs", "2022", "2022abs", ""]
 
-    # Let's check the args to make sure we should do this request.
+    # Always treat 'date' as the date and 'param' as the type
     try:
-        allowed_params.index(args[0])
+        fetch_date = datetime.strptime(date, "%B %d, %Y")
+    except ValueError:
+        await interaction.response.send_message(
+            "Invalid date format! Please use e.g. March 1, 2024.",
+            ephemeral=True,
+        )
+        checkOldFolders()
+        return
+    fetch_type = param
 
-        if (args[0] == "sig" or args[0] == "life") and args[1] != None:
-            allowed_params.index(args[1])
+    # Validate type
+    if fetch_type not in allowed_params:
+        await interaction.response.send_message(
+            "Incorrect params! Example of proper commands: `/fetch March 1, 2024 tor 2024 12`",
```

Comment:
/fetch param:tor date:March 15, 2025 models:2024abs zulu:18 <- Correct way when you copy from discord, and can be pasted and ran that way. The way it's wrote now mimics old $command style.
```suggestion
            "Incorrect params! Example of a proper command: `/fetch param:tor date:March 15, 2025 models:2024abs zulu:18`",
```